drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,482 @@
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <stdlib.h>
+#include "no_os_i2c.h"
+#include "no_os_util.h"
+#include "no_os_error.h"
+#include "no_os_delay.h"
+#include "no_os_alloc.h"
+#include "max9611.h"
+
+/***************************************************************************//**
+ * @brief Generates slave address based on A1 and A0 pin values
+ * @param a0 - A0 logic level
+ * @param a1 - A1 logic level
+ * @return addr - Computed slave ID
+*******************************************************************************/
+int max9611_addr_gen(enum max9611_addr_lvls a0, enum max9611_addr_lvls a1)
+{
+	int addr;
+
+	switch (a1) {
+	case MAX9611_ZERO_VCC:
+		addr = 0xE0;
+		break;
+	case MAX9611_33_VCC:
+		addr = 0xE8;
+		break;
+	case MAX9611_66_VCC:
+		addr = 0xF0;
+		break;
+	case MAX9611_100_VCC:
+		addr = 0xF8;
+		break;
+	default:
```

Comment:
you should add a default case that returns an Error code if none of the enum values were passed to the function. We prefer returning error codes for functions. I'd store the slave address in a member of the devices structure.

Reply:
Init param passed as additional argument and function to return error code (-EINVAL) in case of invalid inputs.